fluent/src/context.js

```diff
@@ -48,7 +50,10 @@ export class MessageContext {
    * @returns {MessageContext}
    */
   constructor(locales, { functions = {}, useIsolating = true } = {}) {
-    this.locales = Array.isArray(locales) ? locales : [locales];
+    // eslint-disable-next-line no-nested-ternary
+    this.locales = Intl.getCanonicalLocales ?
+      Intl.getCanonicalLocales(locales) :
+      Array.isArray(locales) ? locales : [locales];
```

Comment:
I think `eslint` was right here :) I'd reformat this as:

```js
if (Intl.getCanonicalLocales) {
    this.locales = …
} else {
    this.locales = …
}
```

If you want to keep the nested ternary, I'd suggest the following formatting:

```js
this.locales = Intl.getCanonicalLocales
  ? Intl.getCanonicalLocales(locales)
  : Array.isArray(locales)
    ? locales
    : [locales];
```

Another alternative would be to add a dumb polyfill of `getCanonicalLocales` in `fluent/src/index` which would only make the `isArray` check. You'd then only use `this.locales = Intl.getCanonicalLocales()` here.